configs/projects/puppet-agent.rb

```diff
@@ -47,9 +65,14 @@
         proj.setting(:host_gem, "/opt/pl-build-tools/bin/gem")
       end
     end
+  elsif platform.is_windows?
+    # For windows, we need to ensure we are building for mingw not cygwin
+    platform_triple = platform.platform_triple
+    host = "--host #{platform_triple}"
   end
 
-  proj.setting(:gem_install, "#{proj.host_gem} install --no-rdoc --no-ri --local ")
+  proj.setting(:gem_install, "#{proj.host_gem} install --no-rdoc --no-ri --local --bindir #{proj.bindir} ")
```

Comment:
https://github.com/puppetlabs/puppet-agent/pull/509 makes this change windows specific